packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -112,6 +123,48 @@ const FolderTreeItem = ({
     }
   };
 
+  const handleKeyDown = event => {
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_RETURN:
+      case KeyCode.KEY_SPACE:
+        event.preventDefault();
+
+        if (isFolder) {
+          handleToggle(event);
+        } else {
+          onClick(event);
```

Comment:
Shouldn't we return more meaningful information in the onClick event such as information about the item selected?  The person providing the callback function should not have to use the native event to figure that out with our React implementation.

Reply:
While I agree, I'm using the existing implementation of how click is handled. This seems like it could be its own story

https://github.com/cerner/terra-framework/pull/1956/files/ea0913c1bbd0b9a3e889d21308d922ff24573ae9#diff-3d436b479e88c6cb5fa660bfe58c6842c902adf62dc5273b82e5bf7b95353ec7R177